core/src/main/scala/org/http4s/MessageOps.scala

```diff
@@ -87,7 +87,7 @@ trait MessageOps extends Any {
     * @return the `Task` which will generate the T
     */
   final def as[T](implicit decoder: EntityDecoder[T]): Task[T] =
-    attemptAs(decoder).fold(throw _, identity)
+    attemptAs(decoder).fold(Task.fail(_), _.pure[Task]).flatten
```

Comment:
Is there a reason you favored this syntax over that used in [fs2 for fromAttempt](https://github.com/functional-streams-for-scala/fs2/blob/series/1.0/core/shared/src/main/scala/fs2/Task.scala#L369).

```scala
attemptAs(decoder).fold(Task.fail, Task.pure).flatten
```